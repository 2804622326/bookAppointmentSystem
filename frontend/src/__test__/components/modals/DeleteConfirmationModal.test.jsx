import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import DeleteConfirmationModal from '../../../components/modals/DeleteConfirmationModal.jsx';

test('confirms deletion when button clicked', () => {
  const confirm = jest.fn();
  const { asFragment } = render(
    <DeleteConfirmationModal show onHide={() => {}} onConfirm={confirm} itemToDelete="item" />
  );
  fireEvent.click(screen.getByText('Delete'));
  expect(confirm).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
