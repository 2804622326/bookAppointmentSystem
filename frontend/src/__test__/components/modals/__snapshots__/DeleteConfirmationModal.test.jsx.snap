// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirms deletion when button clicked 1`] = `<DocumentFragment />`;
